internal/tiltfile/files.go

```diff
@@ -80,6 +80,19 @@ func (s *tiltfileState) execLocalCmd(t *starlark.Thread, cmd model.Cmd, logOutpu
 		c.Stderr = io.MultiWriter(stderr, logOutput)
 	}
 
+	// if Tilt was invoked with `tilt up --port=XXXXX`, local() calls to use the Tilt API will fail due to trying to
+	// connect to the default port, so explicitly populate the TILT_PORT environment variable if it isn't already
+	hasTiltPort := false
+	for _, e := range c.Env {
```

Comment:
it might make sense to do this in PrepareEnv with the other Env injections?
https://github.com/tilt-dev/tilt/blob/master/pkg/logger/env.go#L19

though I guess you'd have do pass webport to that function...

Reply:
Yeah I thought about that but it seemed weird since that's `logger.PrepareEnv` and pretty specific to xterm type stuff.

I (_very_ briefly) looked at doing it in `Execer` and having `local()` use that but `Execer` is way more oriented to use for the BaDer for daemons and it seemed more headache/risky than it was worth. I think maybe the right thing in the future is to replace the pure `localexec.ExecCmd` with a struct that does the preparation so that it can hold state of things to inject (bonus: you also wouldn't need to pass a logger every time).